x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/helpers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEqual } from 'lodash';
+
+import { Schema } from '../../../../shared/types';
+import { DEFAULT_FIELD_SETTINGS, DISABLED_FIELD_SETTINGS } from '../constants';
+import {
+  FieldResultSetting,
+  FieldResultSettingObject,
+  ServerFieldResultSetting,
+  ServerFieldResultSettingObject,
+} from '../types';
+
+const updateAllFields = (
+  fields: FieldResultSettingObject | ServerFieldResultSettingObject,
+  newValue: FieldResultSetting | {}
+) => {
+  return Object.keys(fields).reduce(
+    (acc, fieldName) => ({ ...acc, [fieldName]: { ...newValue } }),
+    {}
+  );
+};
+
+export const clearAllServerFields = (fields: ServerFieldResultSettingObject) =>
+  updateAllFields(fields, {});
+
+export const clearAllFields = (fields: FieldResultSettingObject) => updateAllFields(fields, {});
+
+export const resetAllServerFields = (fields: ServerFieldResultSettingObject) =>
+  updateAllFields(fields, { raw: {} });
+
+export const resetAllFields = (fields: FieldResultSettingObject) =>
+  updateAllFields(fields, DEFAULT_FIELD_SETTINGS);
+
+export const convertToServerFieldResultSetting = (fieldResultSetting: FieldResultSetting) => {
+  const serverFieldResultSetting: ServerFieldResultSetting = {};
+  if (fieldResultSetting.raw) {
+    serverFieldResultSetting.raw = {};
+    if (fieldResultSetting.rawSize) {
+      serverFieldResultSetting.raw.size = fieldResultSetting.rawSize;
+    }
+  }
+
+  if (fieldResultSetting.snippet) {
+    serverFieldResultSetting.snippet = {};
+    if (fieldResultSetting.snippetFallback) {
+      serverFieldResultSetting.snippet.fallback = fieldResultSetting.snippetFallback;
+    }
+    if (fieldResultSetting.snippetSize) {
+      serverFieldResultSetting.snippet.size = fieldResultSetting.snippetSize;
+    }
+  }
+
+  return serverFieldResultSetting;
+};
+
+export const splitResultFields = (resultFields: FieldResultSettingObject, schema: Schema) => {
```

Comment:
The test file is awesomely organized in a way where the progression of the utility fns make a lot of sense to me. Any chance we can update the utils file to match? In this specific scenario, can we move the `splitResultFields` helper to the bottom of the file?

Reply:
5aa1477f560affb725817a71b2a7da75a8454e07